File: src/main/java/com/enesgunumdogdu/hospital_management/service/ExaminationService.java
package com.enesgunumdogdu.hospital_management.service;

import com.enesgunumdogdu.hospital_management.domain.dto.ExaminationDto;

import java.util.List;

public interface ExaminationService {
    
    List<ExaminationDto> getAllExaminations();
    
    ExaminationDto getExaminationById(Long id);
    
    ExaminationDto createExamination(ExaminationDto examinationDto);
    
    ExaminationDto updateExamination(Long id, ExaminationDto examinationDto);
    
    void deleteExamination(Long id);
}

File: frontend-hospital-management/src/services/examinationService.js
import api from './api'

const examinationService = {
  getAllExaminations() {
    return api.get('/examinations')
  },

  getExaminationById(id) {
    return api.get(`/examinations/${id}`)
  },

  createExamination(examination) {
    return api.post('/examinations', examination)
  },

  updateExamination(id, examination) {
    return api.put(`/examinations/${id}`, examination)
  },

  deleteExamination(id) {
    return api.delete(`/examinations/${id}`)
  }
}

export default examinationService

